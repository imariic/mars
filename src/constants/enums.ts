export enum AppRoutes {
  HOME = "/",
  POSTS = "/app",
  POST_DETAILS = "/post/id",
}
