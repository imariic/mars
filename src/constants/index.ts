export enum ROUTES {
  HOME = "/",
  POSTS = "/app",
  POST_DETAILS = "/post/id",
}

export const AUTH_LOCAL_STORAGE_KEY = "auth";
