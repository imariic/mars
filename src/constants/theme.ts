export const theme = {
  colors: {
    black: "#393635",
    gray: "#686564",
    lightGray: "#969494",
    lighterGray: "#C5C4C3",
    white: "#F3F3F3",
    orange: "#E25033",
    lightOrange: "#E88977",
    lighterOrange: "#ECAA9E",
  },

  dimensions: {
    xsmall: "0.25rem",
    small: "0.5rem",
    medium: "1rem",
    xmedium: "1.5rem",
    large: "2rem",
    xlarge: "2.5rem",
  },
};
