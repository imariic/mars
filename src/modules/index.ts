export * from "./login";
export * from "./post";
export * from "./user";
