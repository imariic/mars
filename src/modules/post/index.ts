export { default as PostsPage } from "./pages/Posts";
