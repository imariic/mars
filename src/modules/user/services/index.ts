import instance from "../../../api";
import { User } from "../../../interfaces";

export const getUsers = async (): Promise<User> => {
  const usersResponse = await instance.get("users");

  return usersResponse.data;
};
