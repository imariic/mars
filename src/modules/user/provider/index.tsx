import React, {
  PropsWithChildren,
  createContext,
  useEffect,
  useState,
} from "react";
import { User } from "interfaces";
import { getUsers } from "modules";

export const UserContext = createContext<User[]>([]);

export const UserProvider: React.FC<PropsWithChildren> = ({ children }) => {
  const [users, setUsers] = useState<User[]>([]);

  useEffect(() => {
    getUsers()
      .then((users) => setUsers(users))
      .catch((e) => console.error("Error", e));
  }, []);

  return <UserContext.Provider value={users}>{children}</UserContext.Provider>;
};
