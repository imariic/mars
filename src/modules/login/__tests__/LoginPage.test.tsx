import { render, screen, waitFor, fireEvent } from "@testing-library/react";
import { AppRoutes } from "constants/enums";
import { theme } from "constants/theme";
import { LoginPage } from "modules/login";
import { UserProvider } from "modules/user";
import { ThemeProvider } from "styled-components";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUsedNavigate,
}));

const renderComponent = () =>
  render(
    <ThemeProvider theme={theme}>
      <UserProvider>
        <LoginPage />
      </UserProvider>
    </ThemeProvider>
  );

describe("LoginPage", () => {
  it("LoginPage renders correctly", () => {
    renderComponent();

    const login = screen.getByText("Login");

    expect(login).toBeInTheDocument();
  });

  it("Checks login functionality with correct credentials", async () => {
    renderComponent();

    const email = screen.getByPlaceholderText("email");
    const password = screen.getByPlaceholderText("password");
    const button = screen.getByLabelText("submit-button");

    fireEvent.change(email, { target: { value: "ivan@gmail.com" } });
    fireEvent.change(password, { target: { value: "ivan" } });

    await waitFor(() => {
      fireEvent.click(button);
      expect(mockedUsedNavigate).toBeCalledWith(AppRoutes.POSTS);
    });
  });

  it("Checks login functionality with incorrect credentials", async () => {
    renderComponent();

    const email = screen.getByPlaceholderText("email");
    const password = screen.getByPlaceholderText("password");
    const button = screen.getByLabelText("submit-button");

    fireEvent.change(email, { target: { value: "wrong email" } });
    fireEvent.change(password, { target: { value: "wrong password" } });

    await waitFor(() => {
      fireEvent.click(button);
      expect(screen.getByText("Wrong credentials")).toBeInTheDocument();
    });
  });
});
