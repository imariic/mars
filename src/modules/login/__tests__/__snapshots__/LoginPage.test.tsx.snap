// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage LoginPage renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-jTQCzO ggivrb"
      >
        <form
          class="sc-guDLey ezlZhe"
          method="POST"
        >
          <h1
            class="sc-dmyCSP edqSwp"
          >
            Login
          </h1>
          <input
            class="sc-beySPh gQutrJ"
            id="email"
            label="email"
            name="email"
            placeholder="email"
            type="email"
          />
          <input
            class="sc-beySPh gQutrJ"
            id="password"
            label="password"
            name="password"
            placeholder="password"
            type="password"
          />
          <input
            aria-label="submit-button"
            class="sc-hLQSwg jQSmTT"
            type="submit"
          />
          <p
            class="sc-eDLKkx hTbvMA"
          />
        </form>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="sc-jTQCzO ggivrb"
    >
      <form
        class="sc-guDLey ezlZhe"
        method="POST"
      >
        <h1
          class="sc-dmyCSP edqSwp"
        >
          Login
        </h1>
        <input
          class="sc-beySPh gQutrJ"
          id="email"
          label="email"
          name="email"
          placeholder="email"
          type="email"
        />
        <input
          class="sc-beySPh gQutrJ"
          id="password"
          label="password"
          name="password"
          placeholder="password"
          type="password"
        />
        <input
          aria-label="submit-button"
          class="sc-hLQSwg jQSmTT"
          type="submit"
        />
        <p
          class="sc-eDLKkx hTbvMA"
        />
      </form>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
